db/warehouse/migrate/20230428155418_add_internal_systems_part3.rb

```diff
@@ -0,0 +1,29 @@
+class AddInternalSystemsPart3 < ActiveRecord::Migration[6.1]
+  def up
+    HmisExternalApis::InboundApiConfiguration.find_each do |conf|
+      internal_system = HmisExternalApis::InternalSystem.where(name: conf.internal_system_name).first_or_initialize
+      if internal_system.new_record?
+        internal_system.active = false
+      end
+      conf.internal_system = internal_system
+      conf.save!
+    end
+
+    remove_column :inbound_api_configurations, :internal_system_name
+  end
+
+  def down
+      add_column :inbound_api_configurations, :internal_system_name, :string
+
+      # The find_each and/or update in ruby seems to be in a different
+      # connection/transaction. Happy to refactor if somebody can explain why.
```

Comment:
Not sure if this is the problem, but you can only talk to one database in a migration (not sure why.)  So if this somehow calls out to the app_db, then it will fail.